src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs

```diff
@@ -0,0 +1,973 @@
+    /// </summary>
+    public readonly struct TimeOnly : IComparable, IComparable<TimeOnly>, IEquatable<TimeOnly>, IFormattable, ISpanFormattable
+    {
+        // represent the number of ticks map to the time of the day. 1 ticks = 100-nanosecond in time measurements.
+        private readonly long _ticks;
+
+        // MinTimeTicks is the ticks for the midnight time 00:00:00.000 AM
+        private const long MinTimeTicks = 0;
+
+        // MaxTimeTicks is the max tick value for the time in the day. It is calculated using DateTime.Today.AddTicks(-1).TimeOfDay.Ticks.
+        private const long MaxTimeTicks = 863999999999;
+
+        /// <summary>
+        /// Represents the smallest possible value of TimeOnly.
+        /// </summary>
+        public static TimeOnly MinValue => new TimeOnly((ulong) MinTimeTicks);
+
+        /// <summary>
+        /// Represents the largest possible value of TimeOnly.
+        /// </summary>
+        public static TimeOnly MaxValue => new TimeOnly((ulong) MaxTimeTicks);
+
+        /// <summary>
+        /// Initializes a new instance of the timeOnly structure to the specified hour and the minute.
+        /// </summary>
+        /// <param name="hour">The hours (0 through 23).</param>
+        /// <param name="minute">The minutes (0 through 59).</param>
+        public TimeOnly(int hour, int minute) : this(DateTime.TimeToTicks(hour, minute, 0, 0)) {}
+
+        /// <summary>
+        /// Initializes a new instance of the timeOnly structure to the specified hour, minute, and second.
+        /// </summary>
+        /// <param name="hour">The hours (0 through 23).</param>
+        /// <param name="minute">The minutes (0 through 59).</param>
+        /// <param name="second">The seconds (0 through 59).</param>
+        public TimeOnly(int hour, int minute, int second) : this(DateTime.TimeToTicks(hour, minute, second, 0)) {}
+
+        /// <summary>
+        /// Initializes a new instance of the timeOnly structure to the specified hour, minute, second, and millisecond.
+        /// </summary>
+        /// <param name="hour">The hours (0 through 23).</param>
+        /// <param name="minute">The minutes (0 through 59).</param>
+        /// <param name="second">The seconds (0 through 59).</param>
+        /// <param name="millisecond">The millisecond (0 through 999).</param>
+        public TimeOnly(int hour, int minute, int second, int millisecond) : this(DateTime.TimeToTicks(hour, minute, second, millisecond)) {}
+
+        /// <summary>
+        /// Initializes a new instance of the TimeOnly structure using a specified number of ticks.
+        /// </summary>
+        /// <param name="ticks">A time of day expressed in the number of 100-nanosecond units since 00:00:00.0000000.</param>
+        public TimeOnly(long ticks)
+        {
+            if ((ulong)ticks > MaxTimeTicks)
+            {
+                throw new ArgumentOutOfRangeException(nameof(ticks), SR.ArgumentOutOfRange_TimeOnlyBadTicks);
+            }
```

Comment:
> but the check allows -MaxTimeTicks through MaxTimeTicks .

This is not true. I am using `ulong` in my comparison which cannot go negative. I am doing that to have it in one check instead of 2 checks.